### ISSUE TITLE
Statically  & dynamically error on a method with duplicate argument name

### ISSUE THREAD

**Issue by fzakaria** [sorbet/sorbet#2273]
#### Error

```bash
[@ @ /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/signature.rb:85]:
NoMethodError: undefined method `+' for nil:NilClass
    @ @ /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/signature.rb:85:in `block in initialize'
    org/jruby/RubyArray.java:2813:in `zip'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/signature.rb:71:in `initialize'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/_methods.rb:295:in `build_sig'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/_methods.rb:268:in `run_sig'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/_methods.rb:200:in `block in _on_method_added'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/_methods.rb:364:in `run_sig_block_for_key'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/_methods.rb:348:in `run_sig_block_for_method'
    /jenkins/.rvm/gems/jruby-9.1.17.0@looker/gems/sorbet-runtime-0.4.5045/lib/types/private/methods/_methods.rb:224:in `block in bar'
```
#### Input

```ruby
class A
  extend T::Sig

  sig {params(_: Integer, _: Integer ).returns(String)}
  def self.bar(_, _)
    ""
  end
end
```

#### Observed output

The `sorbet.run` doesn't fail because this is a `sorbet-runtime` error.
(It would be great to run code and test sorbet-runtime)

I believe the error to be related to having the same argument name twice.

**Comment by DarkDimius** [comment]
We're unlikely to support methods with multiple arguments with _ as name.
That said, it would be nice to have a better error message

**Comment by fzakaria** [comment]
Makes sense -- our codebase is large so there's the pattern a lot of having more than 1 with underscore for unused methods.
(A form of documentation allowing the reader to know the variables are unused)

**Comment by jez** [comment]
Fixed in #3345
